src/server/creatures/helpers/schemas/file-validator.test.js

```diff
@@ -0,0 +1,112 @@
+import Joi from 'joi'
+import { fileValidator } from '~/src/server/creatures/helpers/schemas/file-validator'
+
+describe('#file-validator', () => {
+  test('custom file validator works when theres no errors', () => {
+    const payload = {
+      fileId: '8b2acda5-4f47-4fa7-bccc-2476f5e3afaa',
+      actualContentType: 'image/jpeg',
+      filename: 'test.jpg',
+      contentType: 'image/jpeg',
+      s3Key:
+        '67a59856-6195-40be-a49b-5d9612a740f8/8b2acda5-4f47-4fa7-bccc-2476f5e3afaa',
+      s3Bucket: 'cdp-example-node-frontend',
+      fileStatus: 'complete',
+      contentLength: 290618
+    }
+
+    const { value, error } = fileValidator.file().validate(payload)
+
+    expect(value).toBe(payload)
+    expect(error).toBeUndefined()
+  })
+
+  test('fails when there is an error from the uploader', () => {
+    const payload = {
+      fileId: '8b2acda5-4f47-4fa7-bccc-2476f5e3afaa',
+      actualContentType: 'image/jpeg',
+      filename: 'test.jpg',
+      contentType: 'image/jpeg',
+      fileStatus: 'complete',
+      contentLength: 290618,
+      hasError: true,
+      errorMessage: 'a fake problem'
+    }
+
+    const { error } = fileValidator.file().validate(payload)
+
+    expect(error.details[0].message).toEqual('a fake problem')
+  })
+
+  test('we can scan an array of files where one fails', () => {
+    const payloadOk = {
+      fileId: '8b2acda5-4f47-4fa7-bccc-2476f5e3afaa',
+      actualContentType: 'image/jpeg',
+      filename: 'york-food-recipt.jpg',
+      contentType: 'image/jpeg',
+      s3Key:
+        '67a59856-6195-40be-a49b-5d9612a740f8/8b2acda5-4f47-4fa7-bccc-2476f5e3afaa',
+      s3Bucket: 'cdp-example-node-frontend',
+      fileStatus: 'complete',
+      contentLength: 290618
+    }
+
+    const payloadBad = {
+      fileId: '8b2acda5-4f47-4fa7-bccc-2476f5e3afaa',
+      actualContentType: 'image/jpeg',
+      filename: 'york-food-recipt.jpg',
+      contentType: 'image/jpeg',
+      s3Key:
+        '67a59856-6195-40be-a49b-5d9612a740f8/8b2acda5-4f47-4fa7-bccc-2476f5e3afaa',
+      s3Bucket: 'cdp-example-node-frontend',
+      fileStatus: 'complete',
+      contentLength: 290618,
+      hasError: true,
+      errorMessage: 'The selected file has a virus.'
+    }
+
+    const { error } = Joi.array()
+      .items(fileValidator.file())
+      .validate([payloadOk, payloadBad])
+
+    expect(error.details[0].message).toEqual('The selected file has a virus.')
+  })
+
+  test('we can scan an array of files where one fails where it could also be an object', () => {
```

Comment:
What is this test doing? Some of these test descriptions can be simplified. It becomes really obvious in the cli or ci that these are important. So you can instantly see what the issue is.

I always tended to start with a `Should...`

With the above in mind its clear to see that `'we can scan an array of files where one fails where it could also be an object'` needs a bit of love to make it simpler to understand what this test is testing